### PR TITLE
Feature split progress by organization

### PR DESCRIPTION
**Pull request by julian-berbel**
## :dart: Goal
To split progress by organization! :sparkles: 
## :memo: Details
Assignments will now only be searched for in the requested organization.

The `skip_unique_name_validation` trait in the `another_test_organization` factory would allow the organization object to be created but it would still prevent the organization from being persisted, which would break a test.

As for the rest of the tests, they are expected behavior changes!

## :warning: Dependencies
None.
## :back: Backwards compatibility
Yes.
## :soon: Future work
migration, and lots of testing!

### CODE REVIEW

**Review by luchotc — APPROVED** [review]
:heart:

File: app/models/concerns/with_assignments.rb [hunk 1]
```diff
@@ -20,8 +20,8 @@ def has_messages_for?(user)
 
   # TODO: When the organization is used in this one, please change guide.pending_exercises
   # TODO: Please do the same on WithAssignmentsBatch
-  def find_assignment_for(user, _organization)
```
Comment (flbulgarelli) [inline app/models/concerns/with_assignments.rb:23]: It is time to fix those TO-DOs, isn't it?


I think that everything will break if we don't change `WithBatchAssignments` too.

Comment (julian-berbel) [reply app/models/concerns/with_assignments.rb:23]: Noticed that during the integration test! fixed!

File: app/models/concerns/with_assignments.rb [hunk 2]
```diff
@@ -19,7 +19,6 @@ def has_messages_for?(user)
   end
 
   # TODO: When the organization is used in this one, please change guide.pending_exercises
```
Comment (julian-berbel) [inline app/models/concerns/with_assignments.rb:21]: This second TODO shall be addressed in another PR!

Comment (flbulgarelli) [reply app/models/concerns/with_assignments.rb:21]: Great!

Comment (luchotc) [reply app/models/concerns/with_assignments.rb:21]: Awesome!

File: spec/models/guide_spec.rb [hunk 3]
```diff
@@ -140,7 +139,7 @@
     end
 
     context 'when progress is in more than one organization' do
-      let(:another_organization) { create(:another_test_organization) }
+      let(:another_organization) { create(:another_test_organization, book: test_organization.book) }
```
Comment (felipecalvo) [inline spec/models/guide_spec.rb:142]: Should we add the following test here? Is it relevant?

https://github.com/mumuki/mumuki-domain/commit/5e7759a1fc27e2e1025b7f6ac5b3ddef7b354b56#diff-852e850ded5e5f33c7b8f853db8fbcd980b09b1c93978c29c67facfda1925016R90-R93

It verifies that progress is deleted only on the corresponding organization and not in others, which I think we're missing in this PR (and it's the intended behavior... right? :laughing:)

I found this on a 2019 branch back when we first started this

**Review by felipecalvo — APPROVED** [review]
:100: :+1:

File: spec/models/guide_spec.rb (hunk 3 above)
Comment (felipecalvo) [reply spec/models/guide_spec.rb:142]: Though I guess we should also update this in Laboratory then: https://github.com/mumuki/mumuki-laboratory/commit/e046bc14160f2506f60e844c2a049826da7de1d6

Comment (julian-berbel) [reply spec/models/guide_spec.rb:142]: > Though I guess we should also update this in Laboratory then: mumuki/mumuki-laboratory@e046bc1

that method defaults to current organization so it should be alright either way! still, it would be the responsible thing to add that :grin:
